src/Shell/GettextShell.php

```diff
@@ -180,50 +172,92 @@ public function update(): int
      *
      * @return void
      */
-    private function setupPaths(): void
+    protected function setupPaths(): void
     {
-        if (isset($this->params['plugin'])) {
-            $plugin = (string)$this->params['plugin'];
-            $paths = [
-                Plugin::classPath($plugin),
-                Plugin::configPath($plugin),
-            ];
-            $this->templatePaths = array_merge($paths, App::path(View::NAME_TEMPLATE, $plugin));
-            $this->defaultDomain = $plugin;
-            $localesPaths = (array)Configure::read('App.paths.locales');
-            foreach ($localesPaths as $path) {
-                if (strpos($path, sprintf('%s%s%s', DS, $plugin, DS)) > 0) {
-                    $this->localePath = $path;
-                    break;
-                }
-            }
+        $version = version_compare(Configure::version(), '4.0.0', '>=') ? 4 : 3;
+        $templateKeyName = $version === 4 ? 'templates' : 'Template';
+        $localeKeyName = $version === 4 ? 'locales' : 'Locale';
```

Comment:
default paths for different versions (3 or 4) are correctly handled.
but... in cakephp you can customize paths (in apps and plugins too) by changing configuration `App.paths`. So... this should work in the most part of scenarios, but it won't when a plugin has a different template or locale path.